src/mesh/boundary_standard.cxx

```diff
@@ -546,6 +551,8 @@ void BoundaryDirichlet::apply_ddt(Field2D &f) {
 }
 
 void BoundaryDirichlet::apply_ddt(Field3D &f) {
+  Mesh* mesh = bndry->localmesh;
+  ASSERT1(mesh = f.getMesh());
```

Comment:
If we require `bndry->localmesh == f.getMesh()` can we remove the `localmesh` member of `bndry` or do we need it for something else?

Reply:
`BoundaryRegion::localmesh` is used in several places in the initialization of `BoundaryRegion`s. At that point they do not have a `Field`, so I think do need their own `localmesh`. It may well be that we can get rid of the requirement as part of the boundary re-design, e.g. by using `Region`s.